### ISSUE TITLE
No error emitted for `dict(*{'A': 10})`, which fails at runtime

### ISSUE THREAD

**Issue by hyperkai** [astral-sh/ty#2684]
### Summary

*Memo:
- python test.py
- ty check test.py
- Python 3.14.0
- ty 0.0.7

PVM gets the error with `dict(*{'A': 10})` but ty doesn't get error as shown below:

```python
                     # python # ty
v = dict(*{'A': 10}) # Error  # No error
```

**Comment by carljm** [comment]
I guess we check that the dictionary is iterable (it is), but we don't check if the result of that iteration unpacks to something the dict constructor actually accepts?

**Comment by AlexWaygood** [comment]
Mypy and pyright also do not emit an error on this snippet, but I'm not sure why -- none of the `dict.__init__` [overloads](https://github.com/python/typeshed/blob/fa659b1def704dea3dc8e25c7857b23eac69df4d/stdlib/builtins.pyi#L1178-L1206) seem to apply here? Could typeshed's `dict.__new__` signature (which is much more permissive) be getting in the way?

**Comment by dhruvmanila** [comment]
I think this might be related to #1825 

The type checking step filters out all overloads except
1. (0) because an unpacking could lead to zero elements
2. (1) because same as previous point and `**kwargs` is an optional parameter

```
  2026-02-02T05:44:56.924279Z TRACE ty_python_semantic::types::call::bind: after step 2, matching_overload_index: Multiple([0, 1])
    at crates/ty_python_semantic/src/types/call/bind.rs:1742 on ThreadId(2)
    in ty_python_semantic::types::call::bind::CallableBinding::check_types with arguments=(self: dict[_KT@dict, _VT@dict], *dict[Unknown | str, Unknown | int]) signature=Overload[[_KT, _VT]() -> None, [_KT, _VT](**kwargs: _VT) -> None, [_KT, _VT](map: SupportsKeysAndGetItem[_KT, _VT], /) -> None, [_KT, _VT](map: SupportsKeysAndGetItem[str, _VT], /, **kwargs: _VT) -> None, [_KT, _VT](iterable: Iterable[tuple[_KT, _VT]], /) -> None, [_KT, _VT](iterable: Iterable[tuple[str, _VT]], /, **kwargs: _VT) -> None, [_KT, _VT](iterable: Iterable[list[str]], /) -> None, [_KT, _VT](iterable: Iterable[list[bytes]], /) -> None]
    in ty_python_semantic::types::infer::infer_scope_types with scope=Id(1000) file=File(System("/Users/dhruv/playground/ty/bug-report/main.py"))
    in ty_python_semantic::types::check_types with file=File(System("/Users/dhruv/playground/ty/bug-report/main.py"))

  2026-02-02T05:44:56.924304Z TRACE ty_python_semantic::types::call::bind: after step 4, matching_overload_index: Multiple([0, 1])
    at crates/ty_python_semantic/src/types/call/bind.rs:1760 on ThreadId(2)
    in ty_python_semantic::types::call::bind::CallableBinding::check_types with arguments=(self: dict[_KT@dict, _VT@dict], *dict[Unknown | str, Unknown | int]) signature=Overload[[_KT, _VT]() -> None, [_KT, _VT](**kwargs: _VT) -> None, [_KT, _VT](map: SupportsKeysAndGetItem[_KT, _VT], /) -> None, [_KT, _VT](map: SupportsKeysAndGetItem[str, _VT], /, **kwargs: _VT) -> None, [_KT, _VT](iterable: Iterable[tuple[_KT, _VT]], /) -> None, [_KT, _VT](iterable: Iterable[tuple[str, _VT]], /, **kwargs: _VT) -> None, [_KT, _VT](iterable: Iterable[list[str]], /) -> None, [_KT, _VT](iterable: Iterable[list[bytes]], /) -> None]
    in ty_python_semantic::types::infer::infer_scope_types with scope=Id(1000) file=File(System("/Users/dhruv/playground/ty/bug-report/main.py"))
    in ty_python_semantic::types::check_types with file=File(System("/Users/dhruv/playground/ty/bug-report/main.py"))

  2026-02-02T05:44:56.924428Z TRACE ty_python_semantic::types::call::bind: after step 5, matching_overload_index: Single(0)
    at crates/ty_python_semantic/src/types/call/bind.rs:1785 on ThreadId(2)
    in ty_python_semantic::types::call::bind::CallableBinding::check_types with arguments=(self: dict[_KT@dict, _VT@dict], *dict[Unknown | str, Unknown | int]) signature=Overload[[_KT, _VT]() -> None, [_KT, _VT](**kwargs: _VT) -> None, [_KT, _VT](map: SupportsKeysAndGetItem[_KT, _VT], /) -> None, [_KT, _VT](map: SupportsKeysAndGetItem[str, _VT], /, **kwargs: _VT) -> None, [_KT, _VT](iterable: Iterable[tuple[_KT, _VT]], /) -> None, [_KT, _VT](iterable: Iterable[tuple[str, _VT]], /, **kwargs: _VT) -> None, [_KT, _VT](iterable: Iterable[list[str]], /) -> None, [_KT, _VT](iterable: Iterable[list[bytes]], /) -> None]
    in ty_python_semantic::types::infer::infer_scope_types with scope=Id(1000) file=File(System("/Users/dhruv/playground/ty/bug-report/main.py"))
    in ty_python_semantic::types::check_types with file=File(System("/Users/dhruv/playground/ty/bug-report/main.py"))
```
